### PR TITLE
Remove serialization patch & add a few more tests.

### PR DESCRIPTION
**Pull request by ailzhang**
This PR covers test cases for https://github.com/pytorch/pytorch/pull/25882 and should be landed only after PT PR is landed.

### CODE REVIEW

**Comment by dlibenzi** [comment]
How are we going to add the alias-check?

**Comment by ailzhang** [comment]
Oh I'll override `torch.save` for it.

**Comment by asuhan** [comment]
What is the long term plan for the `torch/serialization.py` patch?

File: test/test_operations.py [hunk 1]
```diff
@@ -853,6 +853,9 @@ def __init__(self, x, y):
     nested = Nested(b, c)
     self.assertRaises(RuntimeError, lambda: xm.check_view_sharing(nested))
 
+    tmpfile = tempfile.mktemp()
+    self.assertRaises(RuntimeError, lambda: torch.save([b, c], tmpfile))
```
Comment (dlibenzi) [inline test/test_operations.py]: This leaves the file around?

**Comment by ailzhang** [comment]
@asuhan Long term we probably want to try to see if we can manage to preserve the views in serialization first.

File: test/test_operations.py (hunk 1 above)
Comment (ailzhang) [reply test/test_operations.py]: Hmmmm this API doesn't create new file so it's fine. 
https://docs.python.org/2/library/tempfile.html#tempfile.mktemp 
But given the warning in doc I might use another API instead :P

File: test/test_operations.py [hunk 2]
```diff
@@ -853,49 +853,40 @@ def __init__(self, x, y):
     nested = Nested(b, c)
     self.assertRaises(RuntimeError, lambda: xm.check_view_sharing(nested))
 
-    tmpfile = tempfile.mktemp()
-    self.assertRaises(RuntimeError, lambda: torch.save([b, c], tmpfile))
+    with tempfile.TemporaryFile() as tf:
```
Comment (dlibenzi) [inline test/test_operations.py]: I am assuming that `torch.save()` takes file pointers on top of file paths, right?

Comment (ailzhang) [reply test/test_operations.py]: Yea it takes both ` a file-like object (has to implement write and flush) or a string containing a file name`.

**Comment by ailzhang** [comment]
https://github.com/pytorch/pytorch/pull/27041 has been merged. Re-trigger circleCI before merging.

**Comment by ailzhang** [comment]
Passed (not sure why github status is not updated), merging :D

**Pull request merged**